app/components/Upload.js

```diff
@@ -69,6 +79,28 @@ export default class Upload extends Component {
     this.onBlockModeInputChange = this.onBlockModeInputChange.bind(this);
     this.onCareLinkInputChange = this.onCareLinkInputChange.bind(this);
     this.onMedtronicSerialNumberInputChange = this.onMedtronicSerialNumberInputChange.bind(this);
+    this.onMedtronicSerialNumberRememberChange = this.onMedtronicSerialNumberRememberChange.bind(this);
+
+    this.populateRememberedSerialNumber();
+  }
+
+  componentWillMount() {
```

Comment:
@krystophv - I added this `componentWillMount` section after your review. Are you happy with this?